### DISCUSSION TITLE
Feature Request: dashboard filter checkbox to exclude in-progress period from time series charts

### DISCUSSION THREAD

**Discussion by sfirke** [apache/superset#30644] [Ideas]
**PROBLEM**

A common challenge in time-series charts -- particularly big number with trendline as it highlights change from the prior period -- is that of the last period being incomplete.  For instance, a chart showing total monthly sales will usually look like a drop in the current month, if that month is incomplete.

I get around this by using an advanced time filter for each chart where the end of the range is `DATETRUNC(DATETIME("now"), month)`.  But this has downsides:
- It's clunky and error-prone to apply to each chart
- It requires hard-coding the time granularity.  If my user sets the dashboard time granularity to week or year they will again be seeing an incomplete final period that is not apples-to-apples with the other periods.

**SUGGESTED SOLUTION**
Have a checkbox that says `Exclude in-progress time period` or similar.  I think it would be a part of the time granularity dashboard filter.  If checked, it would apply this `DATETRUNC` to the relevant queries.  If it's 2024-10-17 and time granularity is set to Month, it would only show data through 2024-09-30.

**Comment by seandiacono** [comment]
This would be super useful! Another suggestion I have is to have the option to compare 2 time periods based on the current period date. For example, when showing a big number trendline for sales data where the current time period is in progress it would be great to be able to compare to the same data from the previous period up until the same hour, day, month, etc..

**Comment by lenngri** [comment]
I would also very much appreciate such a feature! Is there any update on this being implemented?
